packages/spectral/src/types/ComponentRegistry.ts

```diff
@@ -44,7 +53,7 @@ type ComponentReferenceTypeValueMap<
   TMap extends Record<ComponentReferenceType, unknown> = {
     actions: ValueExpression<TValue>;
     triggers: ValueExpression<TValue> | ConfigVarExpression;
-    dataSources: ValueExpression<TValue> | ConfigVarExpression;
+    dataSources: ValueExpression<TValue> | ConfigVarExpression | TemplateExpression;
```

Comment:
Is adding `TemplateExpression` here accurate? I understand adding it to the `values` collections below but this seems more about referencing data sources than what input expressions they support.

Reply:
Without it, TypeScript complains that you cannot specify a "template" for the value of a referenced data source's input

<img width="1165" alt="image" src="https://github.com/user-attachments/assets/15585ad2-856b-48f9-ad12-80266c671257" />

Reply:
Ah, interesting - thanks for the explainer!